openmc/mgxs_library.py

```diff
@@ -457,102 +340,262 @@ def use_chi(self, use_chi):
         check_type('use_chi', use_chi, bool)
         self._use_chi = use_chi
 
-    @total.setter
-    def total(self, total):
+    def add_temperature(self, temperature):
+        """This method re-sizes the attributes of this XSdata object so that it
+        can accomodate an additional temperature.  Note that the set_* methods
+        will still need to be executed.
+
+        Parameters
+        ----------
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset.
+
+        """
+        check_type('temperature', temperature, Real)
+
+        temp_store = self.temperatures.tolist().append(temperature)
+        self.temperatures = temp_store
+
+        self._total.append(None)
+        self._absorption.append(None)
+        self._scatter_matrix.append(None)
+        self._multiplicity_matrix.append(None)
+        self._fission.append(None)
+        self._nu_fission.append(None)
+        self._kappa_fission.append(None)
+        self._chi.append(None)
+        self._inverse_velocities.append(None)
+
+    def set_total(self, total, temperature=294.):
```

Comment:
I agree. Though I do recall a day nearly two years ago where I tried to persuade @paulromano to let me keep my beloved `set_*` methods in my initial PR of the Python API. But he convinced me otherwise to use properties everywhere, and my view of the world has changed :-)